clarity on x-extensible-enum usage for events (#807)
zalando/restful-api-guidelines: chapters/compatibility.adoc

* Enum ranges cannot be extended when used for output parameters — clients may
  not be prepared to handle it. However, enum ranges can be extended when used
  for input parameters.
* Use {x-extensible-enum} (see <<112, rule 112>>), if range is used for output parameters and likely to
  be extended with growing functionality. It defines an open list of explicit
  values and clients must be agnostic or provide default behavior for unknown 

>   values (see <<108, rule 118>>).

Reviewer: ```suggestion
  values (see <<108, rule 108>>).
```
I guess a typo?  Rule 118 is about `snake_case`: https://opensource.zalando.com/restful-api-guidelines/#118

    - EMAIL
----

*Note:* {x-extensible-enum} is a proprietary extension of the JSON Schema standard that 

> will be ignored by most tools.

Reviewer 2: Does it make sense to mention that the swagger UI tooling is showing it in the latest versions?

Author: done

> is e.g. visible via Swagger UI and ignored by most tools.

Reviewer 2: ```suggestion
is e.g. visible via Swagger UI, but ignored by most other tools.
```

See <<240>> about enum value naming conventions.

Note, {x-extensible-enum} is a different concept than JSON schema `examples` which is 

> just a list of few example values, whereas {x-extensible-enum} defines all valid 

Reviewer 2: ```suggestion
just a list of a few example values, whereas {x-extensible-enum} defines all valid 
```

values (for a specific API and service version) and has the advantage of an extensible 
full type-range specification that is validated by the service.

*Important:* Clients must be prepared for extensions of enums returned with server responses, i.e. 
must implement a fallback / default behavior to handle unknown new enum values -- see <<108>>. 
API owners must take care to extend enums in a compatible way that does not change the 
semantics of already existing enum values, for instance, do not split an old enum value 
into two new enum values. Services should only extend {x-extensible-enum} ranges, and only accept 
and return values listed in the latest API version which needs to be updated first before 

> providing new values -- see also <<107>>. 

Author: done -- pls. check.
